### PR TITLE
os\bluestore\NVMEDevice: print error code in aio_handle()

### PR DESCRIPTION
**Pull request by NancySu05**
os: print error code in NVMEDevice::aio_handle
In the aio_handle() function, error code information is not given in the exception handling after the read, write, and refresh operations, and an error code message needs to be added.

Fixes: https://tracker.ceph.com/issues/41208
Signed-off-by: NancySu05  su_nan@inspur.com

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: src/test/common/test_blkdev_sys_block/sys/dev/block/8:0 [hunk 1]
```diff
@@ -1 +0,0 @@
-../../block/sda
```
Comment (tchaikov) [inline src/test/common/test_blkdev_sys_block/sys/dev/block/8:0]: @NancySu05 could you drop this change?

File: src/test/common/test_blkdev_sys_block/sys/dev/block/9:0 [hunk 2]
```diff
@@ -1 +0,0 @@
-../../block/sdb
```
Comment (tchaikov) [inline src/test/common/test_blkdev_sys_block/sys/dev/block/9:0]: ditto.

File: src/os/bluestore/NVMEDevice.cc [hunk 3]
```diff
@@ -421,7 +421,7 @@ void SharedDriverQueueData::_aio_handle(Task *t, IOContext *ioc)
               ns, qpair, lba_off, lba_count, io_complete, t, 0,
               data_buf_reset_sgl, data_buf_next_sge);
           if (r < 0) {
-            derr << __func__ << " failed to read" << dendl;
+            derr << __func__ << " failed to read, return err: " << cpp_strerror(r) << dendl;
```
Comment (tchaikov) [inline src/os/bluestore/NVMEDevice.cc]: s/, return err//

File: src/test/common/test_blkdev_sys_block/sys/dev/block/8:0 (hunk 1 above)
Comment (NancySu05) [reply src/test/common/test_blkdev_sys_block/sys/dev/block/8:0]: no, just commit NVMEDevice.cc.

File: src/test/common/test_blkdev_sys_block/sys/dev/block/9:0 (hunk 2 above)
Comment (NancySu05) [reply src/test/common/test_blkdev_sys_block/sys/dev/block/9:0]: no, just commit NVMEDevice.cc.

File: src/os/bluestore/NVMEDevice.cc (hunk 3 above)
Comment (NancySu05) [reply src/os/bluestore/NVMEDevice.cc]: ok,

**Comment by tchaikov** [comment]
@NancySu05 please note, the title of a PR is different from that of the commit messages of the commits in it. you might need to revise your commit using `git commit --amend` and repush.

**Comment by tchaikov** [comment]
@NancySu05 the title now reads

> aio_handle, without error code;

please revise it by following the suggestions at https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes

i also left a comment couple days ago regarding to this, but it was not addressed yet. see https://github.com/ceph/ceph/pull/29610#pullrequestreview-273724160

File: src/test/common/test_blkdev_sys_block/sys/dev/block/8:0 (hunk 1 above)
Comment (tchaikov) [reply src/test/common/test_blkdev_sys_block/sys/dev/block/8:0]: this comment is not addressed.

File: src/test/common/test_blkdev_sys_block/sys/dev/block/9:0 (hunk 2 above)
Comment (tchaikov) [reply src/test/common/test_blkdev_sys_block/sys/dev/block/9:0]: this comment is not addressed.

**Comment by tchaikov** [comment]
https://github.com/ceph/ceph/pull/29610#issuecomment-521097871 is not addressed.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-08-15_06:56:12-rados-wip-kefu-testing-2019-08-15-1125-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-08-16_05:52:53-rados-wip-kefu-testing-2019-08-16-0928-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2019-08-16_11:16:42-rados-wip-kefu-testing-2019-08-16-1501-distro-basic-smithi/

**Comment by NancySu05** [comment]
I have updated according to your suggestion. Please review,thanks. @ tchaikov

**Comment by NancySu05** [comment]
I have updated according to your suggestion. Please review,thanks. @tchaikov

**Comment by tchaikov** [comment]
@NancySu05 i don't think any of these comment was addressed.

**Comment by NancySu05** [comment]
> @NancySu05 i don't think any of these comment was addressed.
In the case WRITE_COMMAND of _aio_handle(), when spdk_nvme_cmd_writev() returns r<0, the value of r cannot be confirmed. The types that r may return are EINVAL, ENOMEM, ENXIO, etc. so I need to add the value of print confirmation r to locate what it is error.

**Comment by tchaikov** [comment]
resurrected at #31932

**Pull request closed without merging**